test/gtest/softmaxcrossentropywithlogits.cpp

```diff
@@ -0,0 +1,176 @@
+#include <miopen/env.hpp>
+#include "softmaxcrossentropywithlogits.hpp"
+
+MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLOAT_ARG)
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_ALL)
+
+namespace softmaxcrossentropywithlogits {
+
+std::string GetFloatArg()
+{
+    const auto& tmp = miopen::GetStringEnv(ENV(MIOPEN_TEST_FLOAT_ARG));
+    if(tmp.empty())
+    {
+        return "";
+    }
+    return tmp;
+}
+
+struct SoftmaxCrossEntropyWithLogitsTestFloat : SoftmaxCrossEntropyWithLogitsTest<float>
+{
+};
+
+struct SoftmaxCrossEntropyWithLogitsTestHalf : SoftmaxCrossEntropyWithLogitsTest<half>
+{
+};
+
+struct SoftmaxCrossEntropyWithLogitsTestBFloat16 : SoftmaxCrossEntropyWithLogitsTest<bfloat16>
+{
+};
+
+struct SoftmaxCrossEntropyWithLogitsTestFloatBwd : SoftmaxCrossEntropyWithLogitsTestBwd<float>
+{
+};
+
+struct SoftmaxCrossEntropyWithLogitsTestHalfBwd : SoftmaxCrossEntropyWithLogitsTestBwd<half>
+{
+};
+
+struct SoftmaxCrossEntropyWithLogitsTestBFloat16Bwd : SoftmaxCrossEntropyWithLogitsTestBwd<bfloat16>
+{
+};
+
+} // namespace softmaxcrossentropywithlogits
+using namespace softmaxcrossentropywithlogits;
+
+// FORWARD TEST
+TEST_P(SoftmaxCrossEntropyWithLogitsTestFloat, SoftmaxCrossEntropyWithLogitsTest)
+{
+    if((miopen::IsEnabled(ENV(MIOPEN_TEST_ALL)) && GetFloatArg() == "--float") ||
+       GetFloatArg() == "--testall")
+    {
+        RunTest();
+        Verify();
+    }
+    else
+    {
+        GTEST_SKIP();
+    }
+};
+
+TEST_P(SoftmaxCrossEntropyWithLogitsTestHalf, SoftmaxCrossEntropyWithLogitsTest)
+{
+    if((miopen::IsEnabled(ENV(MIOPEN_TEST_ALL)) && GetFloatArg() == "--half") ||
+       GetFloatArg() == "--testall")
+    {
+        RunTest();
+        Verify();
+    }
+    else
+    {
+        GTEST_SKIP();
+    }
+};
+
+TEST_P(SoftmaxCrossEntropyWithLogitsTestBFloat16, SoftmaxCrossEntropyWithLogitsTest)
+{
+    if((miopen::IsEnabled(ENV(MIOPEN_TEST_ALL)) && GetFloatArg() == "--bfloat16") ||
+       GetFloatArg() == "--testall")
+    {
+        RunTest();
+        Verify();
+    }
+    else
+    {
+        GTEST_SKIP();
+    }
+};
+
+INSTANTIATE_TEST_SUITE_P(SoftmaxCrossEntropyWithLogitsTestSet,
+                         SoftmaxCrossEntropyWithLogitsTestFloat,
+                         testing::ValuesIn(SoftmaxCrossEntropyWithLogitsTestConfigs()));
+INSTANTIATE_TEST_SUITE_P(SoftmaxCrossEntropyWithLogitsTestSet,
+                         SoftmaxCrossEntropyWithLogitsTestHalf,
+                         testing::ValuesIn(SoftmaxCrossEntropyWithLogitsTestConfigs()));
+INSTANTIATE_TEST_SUITE_P(SoftmaxCrossEntropyWithLogitsTestSet,
+                         SoftmaxCrossEntropyWithLogitsTestBFloat16,
+                         testing::ValuesIn(SoftmaxCrossEntropyWithLogitsTestConfigs()));
+
+// BACKWARD TEST
+TEST_P(SoftmaxCrossEntropyWithLogitsTestFloatBwd, SoftmaxCrossEntropyWithLogitsTestBwd)
+{
+    if((miopen::IsEnabled(ENV(MIOPEN_TEST_ALL)) && GetFloatArg() == "--float") ||
+       GetFloatArg() == "--testall")
```

Comment:
AMD said test all under the following conditions:
```miopen::IsUnset(ENV(MIOPEN_TEST_ALL))```
https://github.com/moreh-dev/MIOpen/pull/27#discussion_r1634984024

Reply:
removed it